mesonbuild/mesonlib.py

```diff
@@ -683,6 +683,18 @@ def default_libexecdir():
     # There is no way to auto-detect this, so it must be set at build time
     return 'libexec'
 
+def default_libpkgconfigdir():
+    if is_freebsd():
+        return 'libdata/pkgconfig'
+    else:
+        return default_libdir() + '/pkgconfig'
```

Comment:
This means that if I do `-Dlibdir=something` and I don't `-Dlibpkgconfigdir=something/pkgconfig` then my pc files won't end up in the right place. This is not only unfortunate, but also not backward compatible, as far as I can tell.

Reply:
Yes, it changes the behavior, and it was why I had to patch `run_project_tests.py` and `run_unittests.py` to pass tests. I don't know if meson has backward compatibility requirement, but I think it is confusing to have a default value of an option to depend on the user-specified value of another option. If we do so, users can no longer discover default values with `meson setup --help` command.

Reply:
Yes meson must keep backward compat when possible. This will break every packaging systems that set a different libdir. I'm not convinced we should allow arbitrary path for pkgconfig dirs here. It really feels like complicated way of saying: don't change dir for freebsd.